### ISSUE TITLE
[DISCUSS] Minimum Maven version?

### ISSUE THREAD

**Issue by ctubbsii** [revelc/formatter-maven-plugin#243]
In order to fix a warning with Maven 3.5.0, the prerequisites section was moved to the `maven-plugin/pom.xml`. This is correct, but the property `maven.min-version` was also bumped to 3.5.0. Since that property is used in the `prerequisites`, now the plugin declares that the earliest supported version that the plugin works on is `3.5.0`.

I do not believe this is correct. There is a difference between the minimum version that the plugin supports, and the minimum version the plugin is built with. I don't see any reason the plugin can't work on earlier versions of Maven 3. If necessary, this means that the prerequisites section should be separated from the `maven.min-version` property, which is also used by the enforcer for the build.

If this is not corrected, this means that it will be incorrect for projects to use the current version of this plugin on any version of Maven prior to 3.5.0, and I don't think that was what was intended.

**Comment by hazendaz** [comment]
@ctubbsii I think I misunderstood the purpose of the tag to start with.  Are you saying that if that is set to 3.5.0 and I try to use maven 3.3.9 to build it will stop the plugin run?  If so, that is definetly not the intention.  I was just trying to force build using maven 3.5.0 as you have assumed.

**Comment by ctubbsii** [comment]
Yeah, but it doesn't just stop the plugin from running. It fails the build entirely.

The `maven.min-version` property was serving two purposes. It sounds like you only wanted to set the `maven-enforcer-plugin` to require 3.5.0 when this plugin is built. I'm guessing we still want `prerequisites` for the plugin to be set to `3.0.5` so the plugin can be executed on older versions of Maven. Does that sound right?

The fix is simple, we just remove that property, and stop using it for both things. Instead, we set those two things (the `maven-enforcer-plugin` configuration and the `prerequisites`) separately, rather than with a single property.

But, it will require a new release so that people can use it... which I would like to help with to ensure I understand how to do it.

**Comment by hazendaz** [comment]
+1  Yes, you got it.  Please do fix :)

**Comment by ctubbsii** [comment]
So, I started looking at this... I may be opening a few new issues as follow up... but I wonder if you might be available for a Google Hangouts or Skype call some time, just to discuss potential plans forward. Sometimes voice-to-voice is easier for coordinating/collaborating than written word.

**Comment by ctubbsii** [comment]
I did a release (2.5.0) for this. I noticed something weird when I did: the release:perform task created two staging repos in Nexus. One had only 2 files, one called simply 2.5.0 and the other called LATEST, as `net/revelc/code/formatter/net.revelc.code.formatter.feature/2.5.0/N/2.5.0/2.5.0` and `net/revelc/code/formatter/net.revelc.code.formatter.feature/2.5.0/N/2.5.0/LATEST`

Very strange. oss.sonatype.org actually let me release that repo... not sure exactly how it's going to show in Maven Central when it gets indexed. Hopefully it doesn't break indexing the main artifacts.
